plugin-k8s/src/lib.rs

```diff
@@ -43,7 +54,11 @@ impl AlumetPlugin for K8sPlugin {
 
     fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
         let config = deserialize_config(config).context("invalid config")?;
-        Ok(Box::new(K8sPlugin { config }))
+
+        if let Ok(mut inotify_guard) = INOTIFY_VAR.lock() {
```

Comment:
Instead of using a global variable with an `Arc<Mutex<Option<Whatever`, you should just store the value in the plugin structure.